docs/src/content/reference/tools/client.mdx

```diff
@@ -170,18 +170,37 @@ const sseClient = new MastraMCPClient({
   name: "sse-client",
   server: {
     url: new URL("https://your-mcp-server.com/sse"),
-    // Optional fetch request configuration
+    // Optional fetch request configuration - Note: requestInit alone isn't enough for SSE
     requestInit: {
       headers: {
         Authorization: "Bearer your-token",
       },
     },
+    // Required for SSE connections with custom headers
+    eventSourceInit: {
+      fetch(input: Request | URL | string, init?: RequestInit) {
+        const headers = new Headers(init?.headers || {});
+        headers.set('Authorization', 'Bearer your-token');
+        return fetch(input, {
+          ...init,
+          headers,
+        });
+      },
+    },
```

Comment:
makes me think we should let them add headers and we can do this internally:
```
server: {
  url: new URL(),
  headers: { k: "v" }
}

Reply:
Yeah I was thinking that too, but I'm not sure how much we should handle on our end for this. Plus I'm worried about them doing something different with eventSourceInit that conflicts with us handling it internally.

Reply:
We could show in TS types you can only add the headers key if you don't have an eventSourceInit and requestInit defined

Reply:
Cool added those in